### PR TITLE
Add stpcpy().

### PR DESCRIPTION
**Pull request by SvenMichaelKlose**
Like strcpy() but returning pointer to ending zero of copied string.

### CODE REVIEW

File: include/string.h [hunk 1]
```diff
@@ -52,6 +52,7 @@ char* __fastcall__ strchr (const char* s, int c);
 int __fastcall__ strcmp (const char* s1, const char* s2);
 int __fastcall__ strcoll (const char* s1, const char* s2);
 char* __fastcall__ strcpy (char* dest, const char* src);
+char* __fastcall__ stpcpy (char* dest, const char* src);
```
Comment (mrdudz) [inline include/string.h:55]: The prototype should be moved below to the non standard ones

**Comment by mrdudz** [comment]
Also please add a test for stpcpy

**Comment by mrdudz** [comment]
Oh and... why not write this function in assembly? Should be easy enough - and almost certainly saves a couple bytes :)

File: include/string.h (hunk 1 above)
Comment (SvenMichaelKlose) [reply include/string.h]: Oops! True. Sorted.

**Comment by SvenMichaelKlose** [comment]
Test added but I kindly refuse to re-implement this in assembly. That's a job for the optimizer department. If the optimizer cannot handle stpcpy(), improving it instead must result in a huge overall improvement of the compiler.

**Comment by mrdudz** [comment]
> Test added but I kindly refuse to re-implement this in assembly. That's a job for the optimizer department. If the optimizer cannot handle stpcpy(), improving it instead must result in a huge overall improvement of the compiler's output.

Go ahead :)

**Comment by SvenMichaelKlose** [comment]
Sorry. Having too much on my plate already to dig into that. Am sure there are experts who could do that in a jiff. Would take me ages.

**Comment by mrdudz** [comment]
> Sorry. Having too much on my plate already to dig into that. Am sure there are experts who could do that in a jiff. Would take me ages.

The current number of active developers is zero - which is exactly why writing library functions in assembler makes a lot of sense.

**Comment by SvenMichaelKlose** [comment]
Oops... right... what a pity. Since no-one was missing stpcpy() really, please just have this pulled in and I'll take a look at the optimizer by next week end.

File: libsrc/common/stpcpy.c [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include <string.h>
+
+char * __fastcall__
+stpcpy (char * dst, const char * src)
+{
```
Comment (mrdudz) [inline libsrc/common/stpcpy.c:5]: this should adhere to the codestyle of the rest of the library, ie use
`char* __fastcall__ stpcpy (char* dest, const char* src)`

**Comment by SvenMichaelKlose** [comment]
I've heard they've invented computers to do such things (e.g. GNU indent). :p There you go.

**Comment by mrdudz** [comment]
> I've heard they've invented computers to do such things (e.g. GNU indent). :p There you go.

I haven't seen any such tool yet that would not royally mess up existing formatting - else i would have integrated it long ago. Feel free to provide an indent rule that matches the existing cc65 codestyle for that matter.

merged, thanks!

**Pull request merged**